packages/arb-token-bridge-ui/src/util/withdrawals/helpers.ts

```diff
@@ -97,12 +167,35 @@ export async function attachNodeBlockDeadlineToEvent(
     return event
   }
 
+  const firstExecutableBlockFromCache = getFirstExecutableBlockFromCache({
+    event,
+    l2ChainID
+  })
+
+  if (firstExecutableBlockFromCache) {
+    return {
+      ...event,
+      nodeBlockDeadline: firstExecutableBlockFromCache
+    }
+  }
+
   const messageReader = L2ToL1MessageReader.fromEvent(l1Provider, event)
 
   try {
     const firstExecutableBlock = await messageReader.getFirstExecutableBlock(
       l2Provider
     )
+    const firstExecutableBlockCacheKey = getMessagesCacheKey({
+      event,
+      l2ChainId: l2ChainID
+    })
+
+    if (firstExecutableBlockCacheKey && firstExecutableBlock) {
```

Comment:
Why do we have to check if there's `firstExecutableBlockCacheKey`, shouldn't it always be a `string`?